sparkly/functions.py

```diff
@@ -153,3 +153,45 @@ def _execute_case(accumulator, case):
     result = reduce(_execute_case, cases, F).otherwise(default)
 
     return result
+
+
+def argmax(field, by, condition=None, nullable=False):
```

Comment:
A couple of thoughts on this function:
- `condition` and `nullable` have the same responsibilities, should we just keep `condition` as a more generic way of setting conditions?
- I would vote to simplify naming policy  (`__tmp_argmax__` in one case and `field` in the other). This function returns `Column`, it's up to the user to name it properly. Such behaviour would be consistent with regular `pyspark.sql.functions` (e.g. if you do `F.max` you will get column named as `max(field_name)`). Let's repeat behaviour of the built-in functions